application/lib/service/database_service.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import '../model/user_dto.dart';
+import '../utils/text_strings.dart';
+
+class AuthenticationService {
+  final FirebaseAuth _firebaseAuth = FirebaseAuth.instance;
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+
+  Future<String?> addUserToDatabase(UserDTO user) async {
+    try {
+      QuerySnapshot querySnapshot = await _firestore
+          .collection('users')
+          .where('email', isEqualTo: user.email)
+          .get();
+      if (querySnapshot.docs.isNotEmpty) {
+        return thisEmailIsAlredyUsed;
```

Comment:
Az error üzenetekhez tartozó string konstansok nevébe érdemes beletenni, hogy ez egy error üzenet, tehát az emailAlreadyUsedErrorMessage egy jobb választás lehet.